app/controllers/api/host_aggregates_controller.rb

```diff
@@ -25,6 +25,32 @@ def delete_resource_main_action(_type, host_aggregate, _data = {})
       {:task_id => host_aggregate.delete_aggregate_queue(current_user.userid)}
     end
 
+    def add_host_resource(type, id, data = {})
+      raise if id.nil? || data["host_id"].nil?
```

Comment:
We should raise a useful error message,
```suggestion
      raise BadRequestError, "Must specify and id"    if id.nil?
      raise BadRequestError, "Must specify a host_id" if data["host_id"].nil?
```

Reply:
remove the check for `data["host_id"].nil?` as well???